app/controllers/v1/posts_controller.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+class V1::PostsController < ApplicationController
+  before_action :pundit_user
+  before_action :set_postable, except: [:destroy]
```

Comment:
I noticed that the only function for `@postable` in this context is to pass a conditional. I was wondering if perhaps this could go in a `pundit` class. Or you can abstract that in a concern and make your controller a little bit smaller.

Reply:
Thanks. I was thinking how I could simplify the logic for that. Probably I'll do your first suggestion, the pundit class.

Reply:
Decided to just adjust my private methods `set_postable` and `set_post` to handle the abstraction of some logic. Further abstracted error response generation as well. For now it's the simpler refactor I can manage :) I hope it has remarkably simplified the main actions.